File: top_hot100/0102BinaryTreeLevelOrderTraversal/level_order.go
package levelorder

// TreeNode 二叉树节点
type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func levelOrder(root *TreeNode) [][]int {
	result := make([][]int, 0)
	if root == nil {
		return result
	}
	curLevelNode := make([]*TreeNode, 0)
	curLevelNode = append(curLevelNode, root)
	for {
		if len(curLevelNode) == 0 {
			break
		}
		nextLevelNodes := make([]*TreeNode, 0)
		curLevelValues := make([]int, 0, len(curLevelNode))
		for _, oneNode := range curLevelNode {
			curLevelValues = append(curLevelValues, oneNode.Val)
			if oneNode.Left != nil {
				nextLevelNodes = append(nextLevelNodes, oneNode.Left)
			}
			if oneNode.Right != nil {
				nextLevelNodes = append(nextLevelNodes, oneNode.Right)
			}
		}
		curLevelNode = nextLevelNodes
		result = append(result, curLevelValues)
	}
	return result
}

File: top_hot100/0102BinaryTreeLevelOrderTraversal/level_order.py
# coding: utf-8

class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        result = []
        if root == None:
            return result
        
        cur_level_nodes = [root]
        while True:
            if len(cur_level_nodes) == 0:
                break
            next_level_nodes = []
            cur_level_values = []
            for node in cur_level_nodes:
                cur_level_values.append(node.val)
                if node.left != None:
                    next_level_nodes.append(node.left)
                if node.right != None:
                    next_level_nodes.append(node.right)

            result.append(cur_level_values)
            cur_level_nodes = next_level_nodes
        return result